ssl/ssl_lib.cc

```diff
@@ -586,7 +586,16 @@ SSL_CTX *SSL_CTX_new(const SSL_METHOD *method) {
     return nullptr;
   }
 
-  if (!SSL_CTX_set_strict_cipher_list(ret.get(), SSL_DEFAULT_CIPHER_LIST) ||
+  const bool has_aes_hw = EVP_has_aes_hardware();
```

Comment:
I understand that we do the override detection in `SSL_CTX_set_ciphersuites`, but should we add it here as well?
It seems like that's [the general pattern](https://github.com/search?q=repo%3Aaws%2Faws-lc%20EVP_has_aes_hardware&type=code) of how `has_aes_hw` is written in the SSL layer. It would also make it more obvious what TLS13 ciphersuites are being set here at first glance (if the override is being set).